src/HealthChecks.RavenDB/RavenDBHealthCheck.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Raven.Client.Documents;
+using Raven.Client.ServerWide.Operations;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HealthChecks.RavenDB
+{
+    /// <summary>
+    /// Health check for RavenDB.
+    /// </summary>
+    /// <seealso cref="Microsoft.Extensions.Diagnostics.HealthChecks.IHealthCheck" />
+    public class RavenDBHealthCheck : IHealthCheck
+    {
+        private readonly string _connectionString;
+        private readonly string _specifiedDatabase;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RavenDBHealthCheck"/> class.
+        /// </summary>
+        /// <param name="connectionString">The connection string to RavenDB.</param>
+        /// <param name="databaseName">Name of the database.</param>
+        /// <exception cref="System.ArgumentNullException">connectionString</exception>
+        public RavenDBHealthCheck(string connectionString, string databaseName = default)
+        {
+            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
+            _specifiedDatabase = databaseName;
```

Comment:
IMHO This parameter should not be empty and should be checked, otherwise the health check is gonna trigger to fail all the time.

Reply:
There are scenarios where I do not need to test the existence of a database. Just check the server is available.

Therefore, the parameter `databaseName`  is optional and the code below is tested if is not null.

```CSharp
if (!string.IsNullOrWhiteSpace(_specifiedDatabase)
    && !databases.Contains(_specifiedDatabase, StringComparer.OrdinalIgnoreCase))
{
    return HealthCheckResult.Unhealthy($"RavenDB doesn't contains '{_specifiedDatabase}' database.");
}
else
{
   return HealthCheckResult.Healthy();
}
```

Reply:
All right @Burgyn . Understood